### PR TITLE
fix(BAC-141): addStack endpoint to return Id and data

### PR DESCRIPTION
**Pull request by smartempire007**
#### Task reference
This section contains a link to a corresponding task in **Linear**.
[linear issue link](https://linear.app/team-axle-skript-project/issue/BAC-141/build-endpoint-for-create-stack)
#### Summary
This section contains a brief description of **_what was accomplished_** within the current PR.
fixing addStack endpoint to make sure the endpoint returns ID and the data 
#### Implementation details
This section describes the **_technical aspects_** of the solution, implemented withing the current PR, that are worth mentioning to the reviewers.
image for the test
addStack endpoint test

#### How to test
This section contains a checklist, describing basic scenarios that would help verify that the submitted code works correctly.
#### References 
Any additional information (screenshots, video, diagrams) that would help understand and review the PR.

### CODE REVIEW

File: backend/app/Http/Controllers/StackController.php [hunk 1]
```diff
@@ -4,28 +4,48 @@
...
+            $payload = json_decode($request->getContent(), true);
+    
+            if(!isset($payload["name"])){
+                return $this->sendResponse(true, "expected a valid payload", "invalid payload given.", null, Response::HTTP_BAD_REQUEST);
+            }
+            
+            $id = Str::uuid();
+            $name = $payload["name"];
+            
+            $restStack = Stack::where("name", $name);
```
Comment (dasaint001) [inline backend/app/Http/Controllers/StackController.php:28]: what's this line doing when you are adding a new stack?

File: backend/app/Models/Stack.php [hunk 2]
```diff
@@ -10,6 +10,6 @@ class Stack extends Model
 {
     use HasFactory, Uuids;
 
-    protected $fillable = ["name"];
+    protected $fillable = ['id', 'name'];
```
Comment (dasaint001) [inline backend/app/Models/Stack.php:13]: why are you filling name here? this would be done by the DB.

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/app/Models/Stack.php (hunk 2 above)
Comment (smartempire007) [reply backend/app/Models/Stack.php]: name of stack has to be create by user

File: backend/app/Http/Controllers/StackController.php [hunk 3]
```diff
@@ -4,28 +4,48 @@
...
 use Illuminate\Http\JsonResponse;
 use App\Http\Requests\StackRequest;
+use Illuminate\Support\Str;
 use Symfony\Component\HttpFoundation\Response;
 
 class StackController extends Controller
 {
-    public function addStack(StackRequest $request): JsonResponse
+
+    public function addStack(Request $request)
```
[the remark this answers is not in the document]
Comment (smartempire007) [reply backend/app/Http/Controllers/StackController.php]: done

File: backend/app/Http/Controllers/StackController.php [hunk 4]
```diff
@@ -4,28 +4,48 @@
...
+            $data = [
+                "id"=> $id,
+                "name"=>$name,
+            ];
+
+            Stack::create($data);
+
+            return $this->sendResponse(false,null, "Stack created.", $data, Response::HTTP_CREATED);
+        }  catch (\Exception $e) {
+            return $this->sendResponse(true,'something went wrong creating stack', $e->getMessage(), null, Response::HTTP_INTERNAL_SERVER_ERROR );
```
[the remark this answers is not in the document]
Comment (smartempire007) [reply backend/app/Http/Controllers/StackController.php]: done

File: backend/app/Http/Controllers/StackController.php (hunk 1 above)
Comment (smartempire007) [reply backend/app/Http/Controllers/StackController.php:29]: to ensure no user is duplicating a stack in the database

File: backend/app/Http/Controllers/StackController.php [hunk 5]
```diff
@@ -4,28 +4,48 @@
...
-            }else {
-                return $this->sendResponse(true, 'Stack failed', Response::HTTP_BAD_REQUEST);
+        try{
+            $payload = json_decode($request->getContent(), true);
+    
+            if(!isset($payload["name"])){
+                return $this->sendResponse(true, "expected a valid payload", "invalid payload given.", null, Response::HTTP_BAD_REQUEST);
+            }
+            
+            $id = Str::uuid();
```
[the remark this answers is not in the document]
Comment (smartempire007) [reply backend/app/Http/Controllers/StackController.php:26]: dicodedev said that i should ensure that when a user create a stack that it should return the id and the data created

[the remark this answers is not in the document]
Comment (dasaint001) [reply backend/app/Http/Controllers/StackController.php:26]: this is not how to return the id and the data created

File: backend/app/Http/Controllers/StackController.php (hunk 1 above)
Comment (dasaint001) [reply backend/app/Http/Controllers/StackController.php:29]: ok, but it can be done in a better manner

**Review by dasaint001 — CHANGES_REQUESTED** [review]

**Comment by dasaint001** [comment]
substitute PR opened.

**Pull request closed without merging**